ksp_plugin/vessel.cpp

```diff
@@ -449,24 +467,44 @@ void Vessel::CreateFlightPlan(
     Ephemeris<Barycentric>::GeneralizedAdaptiveStepParameters const&
         flight_plan_generalized_adaptive_step_parameters) {
   auto const flight_plan_start = backstory_->back();
-  flight_plan_ = std::make_unique<FlightPlan>(
+  flight_plans_.emplace_back(make_not_null_unique<FlightPlan>(
       initial_mass,
       /*initial_time=*/flight_plan_start.time,
       /*initial_degrees_of_freedom=*/flight_plan_start.degrees_of_freedom,
       final_time,
       ephemeris_,
       flight_plan_adaptive_step_parameters,
-      flight_plan_generalized_adaptive_step_parameters);
+      flight_plan_generalized_adaptive_step_parameters));
+  selected_flight_plan_ = flight_plans_.size() - 1;
+}
+
+void Vessel::DuplicateFlightPlan() {
+  auto const& original = flight_plans_[selected_flight_plan_];
+  auto const it = flight_plans_.begin() + selected_flight_plan_;
+  if (std::holds_alternative<serialization::FlightPlan>(original)) {
+    flight_plans_.emplace(it, std::get<serialization::FlightPlan>(original));
+  } else if (std::holds_alternative<not_null<std::unique_ptr<FlightPlan>>>(
+                 original)) {
+    std::get<not_null<std::unique_ptr<FlightPlan>>>(original)->WriteToMessage(
```

Comment:
This code is somewhat worrisome as it assumes that the serialized form can be freely copied.  A comment would be welcome.

Reply:
Added a comment; the one we should adjust would probably not be this one anyway.